test/widgets/content_test.dart

```diff
@@ -161,4 +161,33 @@ void main() {
       check(tester.takeException()).isA<AssertionError>();
     });
   });
+
+  group("CodeBlock", () {
+    Future<void> prepareContent(WidgetTester tester, String html) async {
+      await tester.pumpWidget(MaterialApp(home: BlockContentList(nodes: parseContent(html).nodes)));
+      await tester.pump();
+      await tester.pump();
+    }
+
+    testWidgets('with syntax highlighting', (WidgetTester tester) async {
+      // "```dart\nclass A {}\n```"
+      await prepareContent(tester,
+        '<div class="codehilite" data-code-language="Dart"><pre>'
+          '<span></span><code><span class="kd">class</span><span class="w"> </span>'
+          '<span class="nc">A</span><span class="w"> </span><span class="p">{}</span>'
+          '\n</code></pre></div>'
+      );
+
+      expect(find.text('class A {}'), findsOneWidget);
```

Comment:
We're writing our tests with `package:checks` and its `check` function, rather than `expect`. See our README:
https://github.com/zulip/zulip-flutter/blob/3cba0582fc68d875872b0b0394d75eedef109bbb/README.md#writing-tests

For this particular use case, we don't have a great idiom yet; that's #232. You could use the idiom mentioned there, or this would also be OK:
```suggestion
      expect(find.text('class A {}'), findsOneWidget); // TODO(#232): use package:checks
```
i.e., leave a comment that marks clearly that this is something we want to fix (which also helps avoid people taking it as an example for other situations) and that makes it easy to find this when we fix #232 and make a sweep of all the affected spots.

Reply:
ack, now uses `tester.widget()`